### PR TITLE
Fix merge fn closure from pr385

### PR DESCRIPTION
**Pull request by BoOTheFurious**
Hello,
I 've got a use case which need to use stateful closure as merge function. As an old PR refers to my needs (PR#385 1 year old) I have rebase on current master. I have also integrated PR#480 to pass the test.
All works fine. I hope It will be accepted quickly.

Thanks @mrobakowski for the works on the initial PR.

Regards

### CODE REVIEW

**Review by stanislav-tkach — APPROVED** [review]

File: src/merge_operator.rs [hunk 1]
```diff
@@ -61,16 +62,25 @@ use std::mem;
 use std::ptr;
 use std::slice;
 
-pub type MergeFn = fn(&[u8], Option<&[u8]>, &mut MergeOperands) -> Option<Vec<u8>>;
+//pub type MergeFn = fn(&[u8], Option<&[u8]>, &mut MergeOperands) -> Option<Vec<u8>>;
```
Comment (aleksuss) [inline src/merge_operator.rs:65]: Is this comment really needed ?

Comment (BoOTheFurious) [reply src/merge_operator.rs:65]: Obviously not, it can be deleted

Comment (Max-Meldrum) [reply src/merge_operator.rs:65]: As I suspect things may get stuck here. I think this was a suggestion for you to update the PR @BoOTheFurious . But from your comment it seems like you expect someone else to do it 😄

Comment (BoOTheFurious) [reply src/merge_operator.rs]: Sorry… I've done it

File: src/merge_operator.rs [hunk 2]
```diff
@@ -41,7 +41,8 @@
 //!let mut opts = Options::default();
 //!
 //!opts.create_if_missing(true);
-//!opts.set_merge_operator("test operator", concat_merge, None);
+//!opts.set_merge_operator_associative("test operator", concat_merge);
+//!//opts.set_merge_operator("test operator", concat_merge, partial_concat_merge); // if your merge is not associative
```
Comment (aleksuss) [inline src/merge_operator.rs:45]: I suppose this comment could be deleted as well ?

Comment (BoOTheFurious) [reply src/merge_operator.rs]: Oops…
done !

Comment (aleksuss) [reply src/merge_operator.rs]: Thanks.

**Review by aleksuss — APPROVED** [review]

**Pull request merged**